### PR TITLE
Updated Microsoft OAuth topic and code to 3.x

### LINKED ISSUE #13299: Update to 3.0: Microsoft Account external login setup
Microsoft Account external login setup

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a4c264c0-0c5c-298e-bb45-9b4891c8d162
* Version Independent ID: ce69b990-0b4c-abda-cd2d-68f85cd8031e
* Content: [Microsoft Account external login setup with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/microsoft-logins?view=aspnetcore-2.2)
* Content Source: [aspnetcore/security/authentication/social/microsoft-logins.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/security/authentication/social/microsoft-logins.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by wadepickett**
[Internal review](https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/social/microsoft-logins?view=aspnetcore-1.0&branch=pr-en-us-15985)

Updated the following topic and code example to 3.x and fixed some additional items.  Tested updates.

- microsoft-logins.md
- social-code/3.x/StartupMS3x.cs

Fixes #13299

### CODE REVIEW

**Comment by wadepickett** [comment]
@Rick-Anderson, I cannot figure out why new 3.x code example is not showing up in this 3.0 update.  Do you happen see what I am missing?

If you view the page at [Internal review](https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/social/microsoft-logins?view=aspnetcore-1.0&branch=pr-en-us-15985)  and scroll down to the section "Configure Microsoft Account Authentication".   The code doesn't show up.

However the link in the topic matches the validated file path from the build, from what I can see.
Path in topic:
~/security/authentication/social/social-code/3.x/StartupMS3x.cs?name=snippet&highlight=37-41

Validated file path in this review build:
aspnetcore/security/authentication/social/social-code/3.x/StartupMS3x.cs

There are no validation errors.  The snippet highlighting seems to be listed right.

**Comment by Rick-Anderson** [comment]
> and scroll down to the section "Configure Microsoft Account Authentication". The code doesn't show up.

Just use the link https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/social/microsoft-logins?view=aspnetcore-3.0&branch=pr-en-us-15985#configure-microsoft-account-authentication

image

Screen shots are helpful. It's showing the 2.2 version with 3.0 in the version selector

image

**Comment by Rick-Anderson** [comment]
There are no snippet regions in ~/security/authentication/social/social-code/3.x/StartupMS3x.cs so I don't know why any snippet is showing up

**Comment by Rick-Anderson** [comment]
* You have the incorrect URL for internal review URL
* You don't have `#region snippet` in the code, so it won't show any code.. That's expected.

**Comment by Rick-Anderson** [comment]
https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/social/microsoft-logins?view=aspnetcore-3.1&branch=pr-en-us-15985#configure-microsoft-account-authentication

File: aspnetcore/security/authentication/social/microsoft-logins.md [hunk 1]
```diff
@@ -53,19 +55,19 @@ Link sensitive settings like Microsoft `ClientId` and `ClientSecret` to your app
...
 
-See the [MicrosoftAccountOptions](/dotnet/api/microsoft.aspnetcore.builder.microsoftaccountoptions) API reference for more information on configuration options supported by Microsoft Account authentication. This can be used to request different information about the user.
+For more information about configuration options supported by Microsoft Account authentication, see the [MicrosoftAccountOptions](/dotnet/api/microsoft.aspnetcore.builder.microsoftaccountoptions) API reference. This can be used to request different information about the user.
 
 ## Sign in with Microsoft Account
 
-Run the and click **Log in**. An option to sign in with Microsoft appears. When you click on Microsoft, you are redirected to Microsoft for authentication. After signing in with your Microsoft Account (if not already signed in) you will be prompted to let the app access your info:
+Run the application and click **Log in**. An option to sign in with Microsoft appears. When you click on Microsoft, you are redirected to Microsoft for authentication. After signing in with your Microsoft Account, (if not already signed in) you will be prompted to let the app access your info:
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/social/microsoft-logins.md]: Run the app and click **Log in**. An option to sign in with Microsoft appears. When you click on Microsoft, you are redirected to Microsoft for authentication. After signing in with your Microsoft Account,  you will be prompted to let the app access your info:

Comment (Rick-Anderson) [reply aspnetcore/security/authentication/social/microsoft-logins.md]: per loc style guide https://review.docs.microsoft.com/en-us/help/contribute/localization-mt-guidance?branch=master
Avoid parenthetical clauses.

File: aspnetcore/security/authentication/social/microsoft-logins.md [hunk 2]
```diff
@@ -4,25 +4,27 @@ author: rick-anderson
...
 
-If you don't have a Microsoft account, select **Create one**. After signing in you are redirected to the **App registrations** page:
+If you don't have a Microsoft account, select **Create one**. After signing in, you are redirected to the **App registrations** page:
 
 * Select **New registration**
 * Enter a **Name**.
 * Select an option for **Supported account types**.  <!-- Accounts for any org work with MS domain accounts. Most folks probably want the last option, personal MS accounts -->
-* Under **Redirect URI**, enter your development URL with `/signin-microsoft` appended. For example, `https://localhost:44389/signin-microsoft`. The Microsoft authentication scheme configured later in this sample will automatically handle requests at `/signin-microsoft` route to implement the OAuth flow.
+* Under **Redirect URI**, enter your development URL with `/signin-microsoft` appended. For example, `https://localhost:44353/signin-microsoft`. The Microsoft authentication scheme configured later in this sample will automatically handle requests at `/signin-microsoft` route to implement the OAuth flow.
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/social/microsoft-logins.md]: * Under **Redirect URI**, enter your development URL with `/signin-microsoft` appended. For example, `https://localhost:5001/signin-microsoft`. The Microsoft authentication scheme configured later in this sample will automatically handle requests at `/signin-microsoft` route to implement the OAuth flow.

Comment (Rick-Anderson) [reply aspnetcore/security/authentication/social/microsoft-logins.md]: Prefer 5001

File: aspnetcore/security/authentication/social/microsoft-logins.md [hunk 3]
```diff
@@ -53,19 +55,19 @@ Link sensitive settings like Microsoft `ClientId` and `ClientSecret` to your app
 
 ## Configure Microsoft Account Authentication
 
-Add the Microsoft Account service to `Startup.ConfigureServices`:
+Add the Microsoft Account service to the `Startup.ConfigureServices` method in the *Startup.cs* file:
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/social/microsoft-logins.md]: Add the Microsoft Account service to the `Startup.ConfigureServices`:

File: aspnetcore/security/authentication/social/microsoft-logins.md [hunk 4]
```diff
@@ -53,19 +55,19 @@ Link sensitive settings like Microsoft `ClientId` and `ClientSecret` to your app
...
 
-[!code-csharp[](~/security/authentication/social/social-code/StartupMS.cs?name=snippet&highlight=10-14)]
+[!code-csharp[](~/security/authentication/social/social-code/3.x/StartupMS3x.cs?name=snippet&highlight=10-14)]
 
 [!INCLUDE [default settings configuration](includes/default-settings.md)]
 
 [!INCLUDE[](includes/chain-auth-providers.md)]
 
-See the [MicrosoftAccountOptions](/dotnet/api/microsoft.aspnetcore.builder.microsoftaccountoptions) API reference for more information on configuration options supported by Microsoft Account authentication. This can be used to request different information about the user.
+For more information about configuration options supported by Microsoft Account authentication, see the [MicrosoftAccountOptions](/dotnet/api/microsoft.aspnetcore.builder.microsoftaccountoptions) API reference. This can be used to request different information about the user.
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/social/microsoft-logins.md]: For more information about configuration options supported by Microsoft Account authentication, see the [MicrosoftAccountOptions](/dotnet/api/microsoft.aspnetcore.builder.microsoftaccountoptions) API reference. `MicrosoftAccountOptions` can be used to request different information about the user.

Comment (Rick-Anderson) [reply aspnetcore/security/authentication/social/microsoft-logins.md]: Acyrolix should have caught that. Make sure you run Acyrolix.

**Comment by Rick-Anderson** [comment]
@scottaddie  I'm not seeing accept change/accept batch - why doesn't that show up on Wade's PR's?

**Comment by scottaddie** [comment]
@Rick-Anderson Try adding the `suggestion` code fencing. For example:

image

**Comment by Rick-Anderson** [comment]
@scottaddie  it made it code fenced as expected. 

I've noticed all Wade's PR have  don't enable accept suggestion.

File: aspnetcore/security/authentication/social/microsoft-logins.md [hunk 5]
```diff
@@ -4,25 +4,27 @@ author: rick-anderson
 description: This sample demonstrates the integration of Microsoft account user authentication into an existing ASP.NET Core app.
 ms.author: riande
 ms.custom: mvc
-ms.date: 05/11/2019
+ms.date: 12/03/2019
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/social/microsoft-logins.md]: ms.date: 12/5/2019

Comment (Rick-Anderson) [reply aspnetcore/security/authentication/social/microsoft-logins.md]: @scottaddie  try suggesting any line. I did the following only to demo the problem

**Comment by Rick-Anderson** [comment]
@wadepickett  I pushed a commit to this branch as a test, so you'll need to get pull before you do another submit.

**Comment by wadepickett** [comment]
@Rick-Anderson, made changes per each request in review.  Ready to merge on your approval.  Thanks.   There was not a "Add suggestion to batch option" on my end, so I was not able to use that.

**Pull request merged**